codegenerator/cli/src/rescript_types.rs

```diff
@@ -576,6 +685,26 @@ mod tests {
         assert_eq!(type_decl.to_string(), expected);
     }
 
+    #[test]
+    fn type_decl_with_invalid_rescript_field_names_to_string_record() {
+        let type_decl = RescriptTypeDecl::new(
+            "myRecord".to_string(),
+            RescriptTypeExpr::Record(vec![
+                RescriptRecordField::new("module".to_string(), RescriptTypeIdent::Bool),
+                RescriptRecordField::new("".to_string(), RescriptTypeIdent::Bool),
+                RescriptRecordField::new("1".to_string(), RescriptTypeIdent::Bool),
+                RescriptRecordField::new("Capitalized".to_string(), RescriptTypeIdent::Bool),
+                // Invalid characters in the middle are not handled correctly. Still keep the test to pin the behaviour.
+                RescriptRecordField::new("dashed-field".to_string(), RescriptTypeIdent::Bool),
+            ]),
+            vec![],
+        );
+
+        let expected = r#"type myRecord = {@as("module") module_: bool, @as("") _: bool, @as("1") _1: bool, @as("Capitalized") capitalized: bool, dashed-field: bool}"#.to_string();
```

Comment:
What's happening with dash-field? If you want, there's a lib I installed called convert_case which you could transform kebab to camel or snake case for eg.

Reply:
It'll codegen like this `dashed-field: bool` and break during compilation. I think we don't have a case for this yet, so I just ignored it.